R/Belgium.R

```diff
@@ -0,0 +1,208 @@
+    source_data_cols = c("cases_new", "deaths_new"),
+
+    #' @description Set up a table of region codes for clean data
+    #' @importFrom tibble tibble tribble
+    set_region_codes = function() {
+      self$codes_lookup$`1` <- tibble::tibble(
+        level_1_region_code = c("BE-BRU", "BE-VLG", "BE-WAL"),
+        level_1_region = c("Brussels", "Flanders", "Wallonia")
+      )
+      self$codes_lookup$`2` <- tibble::tribble(
+        ~level_2_region_code,    ~level_2_region, ~level_1_region_code,
+                    "BE-VAN",        "Antwerpen",      "BE-VLG",
+                    "BE-WBR",    "BrabantWallon",      "BE-WAL",
+                    "BE-WHT",          "Hainaut",      "BE-WAL",
+                    "BE-WLG",       "Li\u00e8ge",      "BE-WAL",
+                    "BE-VLI",          "Limburg",      "BE-VLG",
+                    "BE-WLX",       "Luxembourg",      "BE-WAL",
+                    "BE-WNA",            "Namur",      "BE-WAL",
+                    "BE-VOV",   "OostVlaanderen",      "BE-VLG",
+                    "BE-VBR",    "VlaamsBrabant",      "BE-VLG",
+                    "BE-VWV",   "WestVlaanderen",      "BE-VLG",
+                    "BE-BRU",         "Brussels",      "BE-BRU"
+        )
+    },
+
+    #' @description directs to either level 1 or level 2 processing based on
+    #' request.
+    #' @importFrom dplyr select mutate
+    #' @importFrom lubridate as_date ymd_hms
+    clean = function() {
+      message_verbose(self$verbose, "Cleaning data")
+
+      # vroom fails to load two lines in the main data set
+      # For now, we filter out the broken lines and replace them
+      # with the following data shim
+
+      self$data$raw$original_main <- self$data$raw$main
+
+      fixed_lines <- tibble::tribble(
+        ~DATE, ~PROVINCE, ~REGION, ~AGEGROUP, ~SEX, ~CASES,
+        "2020-04-22", "Limburg", "Flanders", "50-59", "F", 10,
+        "2021-02-17", "VlaamsBrabant", "Flanders", "10-19", "M", 12
+      ) %>% mutate(DATE = as.Date(DATE), CASES = as.double(CASES))
+
+      self$data$raw$main <-
+        self$data$raw$main %>%
+          filter((REGION %in% self$codes_lookup[[1]]$level_1_region
+                   | is.na(REGION))) %>%
+        bind_rows(fixed_lines)
+
+      if (self$level == "1") {
+        self$clean_level_1()
+      } else if (self$level == "2") {
+        self$clean_level_2()
+      }
+    },
+
+    #' @description Belgium Specific Region Level Data Cleaning
+    # nolint start
+    #' @importFrom dplyr group_by summarise ungroup full_join tally mutate select rename
+    #' @importFrom tidyr replace_na
+    #' @importFrom lubridate ymd
+    # nolint end
+    clean_level_1 = function() {
+      cases_data <- self$data$raw$main %>%
+        select(DATE, REGION, CASES) %>%
+        mutate(
+          DATE = ymd(DATE),
+          CASES = as.numeric(CASES)
+        ) %>%
+        replace_na(list(REGION = "Unknown")) %>%
```

Comment:
Quite a bit of copy pasta in the cleaning code. Worth adding a method to the Belgium class and reusing in this code (+ maybe a cheeky test?)